src/hostapi/wasapi/pa_win_wasapi.c

```diff
@@ -3910,7 +3956,7 @@ static PaError OpenStream( struct PaUtilHostApiRepresentation *hostApi,
         if (stream->in.params.blocking == TRUE)
         {
             UINT32 bufferFrames = ALIGN_NEXT_POW2((stream->in.framesPerHostCallback / WASAPI_PACKETS_PER_INPUT_BUFFER) * 2);
-            UINT32 frameSize    = stream->in.wavexu.ext.Format.nBlockAlign;
+            UINT32 frameSize    = stream->in.wavexu.ext.Format.nBlockAlign / (stream->in.monoMixer != NULL ? 2 : 1);
```

Comment:
Are we dividing the input bytes per frame by two to get the output bytes per frame because we are mixing from stereo to mono?

Reply:
Here ring buffer is serving user with mono data while WASAPI backend is operating with stereo data (`stream->in.wavexu.ext.Format.nBlockAlign`), therefore to avoid using 2x larger buffer size we shrink it to the actual size being served to the user.